Manifold/Term.swift

```diff
@@ -143,80 +129,92 @@ public struct Term: BooleanLiteralConvertible, CustomDebugStringConvertible, Fix
 	}
 
 
+	// MARK: Bound variables
+
+	private func map(transform: (Int, Int) -> Term) -> Term {
+		return zeppo { parents, expression in
+			expression.analysis(
+				ifBound: { transform(parents.count, $0) },
+				ifApplication: { Term.application($0, $1) },
+				ifPi: { Term.pi($0, $1) },
+				ifProjection: { Term.projection($0, $1) },
+				ifSigma: { Term.sigma($0, $1) },
+				ifIf: { Term.`if`($0, then: $1, `else`: $2) },
+				otherwise: const(Term(expression)))
+		} (self)
+	}
```

Comment:
The morphism in action. The case for `bound` variables is able to provide the depth of a given term to its `transform` parameter, enabling shifting and substitution to share most of their implementation.